src/gainmap.c

```diff
@@ -82,8 +80,8 @@ static float avifGetGainMapWeight(float hdrHeadroom, const avifGainMapMetadataDo
         return 0.0f;
     }
     float w = AVIF_CLAMP((hdrHeadroom - baseHdrHeadroom) / (alternateHdrHeadroom - baseHdrHeadroom), 0.0f, 1.0f);
-    if (metadata->backwardDirection) {
-        w *= -1.0f;
+    if (alternateHdrHeadroom < baseHdrHeadroom) {
+        w = -w;
     }
     return w;
```

Comment:
optional: `w` can be `const` and this can be a ternary return:

```
return (alternateHdrHeadroom < baseHdrHeadroom) ? -w : w;
```

Reply:
Done.